src/render/hw/draw/fragment/wgsl_solid_color.cc

```diff
@@ -25,9 +25,15 @@ void WGSLSolidColor::WriteFSUniforms(std::stringstream& ss) const {
 }
 
 void WGSLSolidColor::WriteFSMain(std::stringstream& ss) const {
-  ss << R"(
-  color = vec4<f32>(uColor.rgb * uColor.a, uColor.a);
-)";
+  if (color_.has_value()) {
```

Comment:
If having color or not produces different shader code, the GetShaderName method should return different as well. This is because GetShaderName is part of the PSO key used to find the correct PSO.